src/tests.ts

```diff
@@ -1,11 +1,40 @@
+import * as os from "os";
 import * as vscode from "vscode";
-import { ExecResult, exec, extensionConfiguration } from "./utils";
-import { Tests, DebugEnvironmentConfiguration } from "./types";
+import { ExecResult, exec, extensionConfiguration, getTargetName } from "./utils";
+import { Targets, Test, Tests, DebugEnvironmentConfiguration } from "./types";
 import { getMesonTests, getMesonTargets } from "./introspection";
 import { workspaceState } from "./extension";
 
+// this is far from complete, but should suffice for the
+// "test is made of a single executable is made of a single source file" usecase.
```

Comment:
Are you going to complete this TODO

Reply:
No.
The comment is about tests that consist of multiple TUs, or tests that are not native meson tests (see https://mesonbuild.com/Reference-manual_functions.html#test_protocol ) - I don't have a good idea on what projects using these options look like, as I personally only use single TU tests.
I'm not sure if there is a simple solution for these, and "single TU tests" should cover the majority of users.